tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -396,21 +397,39 @@ SpanBuffer SwapBuffers()
             }
 
             RunSpanSampler(spans);
-
+            int? chunkSamplingPriority = null;
             if (CanComputeStats)
             {
                 spans = _statsAggregator?.ProcessTrace(spans) ?? spans;
                 bool shouldSendTrace = _statsAggregator?.ShouldKeepTrace(spans) ?? true;
                 _statsAggregator?.AddRange(spans);
+                var singleSpanSamplingSpans = new List<Span>(); // TODO maybe we can store this from above?
+
+                for (var i = spans.Offset; i < spans.Count; i++)
```

Comment:
This loop doesn't look right. If the `ArraySegment` has `Offset=10` and `Count=2`, you would want `i` to be `10` first, then `11`. However, the condition `i < spans.Count` will be false and the code inside the loop would never run.

I think this works today because all our `ArraySegment` have `Offset=0`, but would should not rely on that.